lib/IRGen/MetadataLayout.cpp

```diff
@@ -111,7 +111,7 @@ MetadataLayout &IRGenModule::getMetadataLayout(NominalTypeDecl *decl) {
   auto &entry = MetadataLayouts[decl];
   if (!entry) {
     if (auto theClass = dyn_cast<ClassDecl>(decl)) {
-      if (theClass->isForeign())
+      if (theClass->getForeignClassKind() == ClassDecl::ForeignKind::CFType)
```

Comment:
Yikes, thank you. Perhaps we should do a renaming pass here, because it seems like IRGen is using "foreign class" in places that really mean "CF type". That's how foreign classes started, but with the introduction the Clang's `objc_runtime_visible`, the meaning of "foreign" in the AST was generalized.

Reply:
I don't know that the concept of `ForeignKind` really holds its weight at all; the capabilities of CF and runtime-only classes overlap somewhat but both diverge in different ways; the Venn diagram doesn't perfectly overlap. IMO We should probably just split the "is CF" and "is objc_runtime_visible" predicates.